src/transformers/pipelines/fill_mask.py

```diff
@@ -74,8 +83,23 @@ def __init__(
 
         self.check_model_type(TF_MODEL_WITH_LM_HEAD_MAPPING if self.framework == "tf" else MODEL_FOR_MASKED_LM_MAPPING)
         self.top_k = top_k
+        self.targets = targets
+        if self.tokenizer.mask_token_id is None:
+            raise PipelineException(
+                "fill-mask", self.model.base_model_prefix, "The tokenizer does not define a `mask_token`."
+            )
 
-    def ensure_exactly_one_mask_token(self, masked_index: np.ndarray):
+    def get_masked_index(self, input_ids: GenericTensor) -> np.ndarray:
+        if self.framework == "tf":
+            masked_index = tf.where(input_ids == self.tokenizer.mask_token_id).numpy()
+        elif self.framework == "pt":
+            masked_index = torch.nonzero(input_ids == self.tokenizer.mask_token_id, as_tuple=False)
```

Comment:
This returns the index of the masked token within the sequences. If an array of sequences was passed, then this will return a two-dimensional array.

Reply:
Well we shouldn't send 2d arrays here.
Should we assert ?